### PR TITLE
Adds Billing occurrence and interval to subscriptions upsert

### LINKED ISSUE #86: subscription upsert select billing cycle
The user should be able to select interval and occurrence when creating or editing a subscription.  Use the mockups as reference

### Acceptance Criteria
- Occurrence should be a Select list 
- interval is just a numerical with a min value of 1
- pluralize occurrence if interval has a value > 1
- Default interval is 1 and Occurrence is Month
- A png of the page is attached to the Pull request (not the repository code)
- Looks good on a ~1080x2400 (google pixel 5)

### PR DESCRIPTION
**Pull request by ashwin003**
Closes #86 and adds billing occurrence and intervals to subscription upsert page.

image

### CODE REVIEW

File: subtrack.MAUI/Pages/UpsertSubscription.razor [hunk 1]
```diff
@@ -33,6 +33,18 @@
     <div class="form-group" style="padding-top: 10px;">
         <InputTextArea class="form-control" rows="3" placeholder="Description" @bind-Value="@_subscription.Description"></InputTextArea>
     </div>
+    <div style="display: flex;padding-top: 10px;">
```
Comment (chrisK00) [inline subtrack.MAUI/Pages/UpsertSubscription.razor:36]: interval and occurrence should be on the same line as showed in the mockups.

**Alt 1:** Paid every {interval} {occurrence}
**Alt 2:** Every {interval} {occurrence}
**Alt 3:**
Billing Cycle
every {interval} {occurrence}
**Alt 4:**
Billing
every {interval} {occurrence}

File: subtrack.MAUI/Pages/UpsertSubscription.razor [hunk 2]
```diff
@@ -33,6 +33,18 @@
...
+        <label style="white-space: nowrap; padding-top: 5px; padding-right: 10px; font-size: 17px;" for="billingOccurrence">Billing Occurrence </label>
+        <InputSelect @bind-Value="@_subscription.BillingOccurrence" class="form-control" id="billingOccurrence">
```
Comment (chrisK00) [inline subtrack.MAUI/Pages/UpsertSubscription.razor]: occurrence should be pluralized if interval > 1

**Review by chrisK00 — CHANGES_REQUESTED** [review]
Type "closes #86 " in the desc instead so the issue closes automatically when the PR is merged.

streak counter: 4

File: subtrack.MAUI/Pages/UpsertSubscription.razor [hunk 3]
```diff
@@ -33,6 +33,21 @@
...
     </div>
+    <div style="display: flex;padding-top: 10px;">
+        <label style="white-space: nowrap; padding-top: 5px; padding-right: 10px; font-size: 17px;">Billing Cycle every </label>
+    </div>
+    <div class="d-flex flex-column flex-sm-row" style="padding-top: 10px;">
+        <div class="flex-fill me-sm-4 me-md-4 me-lg-4 mb-4">
+            <InputNumber class="form-control" placeholder="Billing Interval" @bind-Value="@_subscription.BillingInterval" id="billingInterval"></InputNumber>
+        </div>
+        <div class="flex-fill">
+            <InputSelect @bind-Value="@_subscription.BillingOccurrence" class="form-control">
```
Comment (chrisK00) [inline subtrack.MAUI/Pages/UpsertSubscription.razor]: Think the comment got lost but dont forgett to check the acceptance criteria/mockups about occurrence pluralization

**Review by chrisK00 — CHANGES_REQUESTED** [review]

File: subtrack.MAUI/Pages/UpsertSubscription.razor (hunk 3 above)
Comment (ashwin003) [reply subtrack.MAUI/Pages/UpsertSubscription.razor]: image

Comment (chrisK00) [reply subtrack.MAUI/Pages/UpsertSubscription.razor]: Nice! 
Move down "every" one line and do the inputs as in the previously specified format

every {1} {month}

Billing cycle altogether can be moved up, below Cost

Comment (ashwin003) [reply subtrack.MAUI/Pages/UpsertSubscription.razor]: image

Comment (chrisK00) [reply subtrack.MAUI/Pages/UpsertSubscription.razor]: Interval could be a lot smaller. Wouldnt expect someone to enter more than 2 numbers

Comment (ashwin003) [reply subtrack.MAUI/Pages/UpsertSubscription.razor]: image

**Review by chrisK00 — APPROVED** [review]

**Pull request merged**